lib/chef/provider/package/homebrew.rb

```diff
@@ -83,9 +87,40 @@ def brew(*args)
         # information, but that is not any more robust than using the
         # command-line interface that returns the same thing.
         #
-        # https://github.com/Homebrew/homebrew/wiki/Querying-Brew
+        # https://docs.brew.sh/Querying-Brew
+        #
+        # @returns [Hash] a hash of package information where the key is the package name
         def brew_info
-          @brew_info ||= Chef::JSONCompat.from_json(brew("info", "--json=v1", new_resource.package_name)).first
+          @brew_info ||= begin
+            command_array = ["info", "--json=v1"].concat Array(new_resource.package_name)
+            # convert the array of hashes into a hash where the key is the package name
+
+            cmd_output = brew_cmd_output(command_array, allow_failure: true)
+            return nil if cmd_output.empty? # empty std_out == bad package queried
```

Comment:
oh this is deeply weird.

so ideally you'd only return a nil for the one package_name that didn't exist.

it is certainly more efficient to only shellout once, but that makes it all strange.

you could return an array of nils  which was of the size of the package_name_array but then it would probably cause the superclass to issue an error about every package being missing.

to do this right you probably need to:

```
if cmd_output.empty?
  package_name_array.each_with_object({}) do |package_name, hsh|
    cmd_output = brew_cmd_output("info", "--json=v1", package_name, allow_failure: true)
    if cmd_output.empty?
      hsh[package_name] = nil
    else
      hsh[json["name"] = Chef::JSONCompat.from_json(cmd_output)
    end
  end
else 
  Hash[Chef::JSONCompat.from_json(cmd_output).collect { |pkg| [pkg["name"], pkg] }]
end
```

Reply:
So that means when it fails, you fall back to iterating one by one through the package_names to determine which one failed and what the versions were for the rest of them.

Reply:
`hsh[package_name] = {}` per my other comments that this should return empty array instead of nil here to make package_info simpler and to contain the damage in this method.

Reply:
I used this with a minor tweak to Chef::JSONCompat.from_json(cmd_output).first since Chef::JSONCompat.from_json(cmd_output) returned and array of hashes.